### PR TITLE
Invalid time conversion fix

### PR DESCRIPTION
**Pull request by jduss4**
When detected time is considered invalid, will not return an "invalid" message to the user

Alters tests with user region / city abbreviations to be more intuitive when reading tests.
Adds passing tests for invalid time, and time ranges around midnight
Adds a skipped test for 24 hour time which accompanies #292 

Closes https://github.com/18F/charlie/issues/257

### CODE REVIEW

File: src/scripts/timezone.test.js [hunk 1]
```diff
@@ -60,10 +60,14 @@ describe("Handy Tau-bot timezone conversions", () => {
     });
 
     slack.getSlackUsersInConversation.mockResolvedValue([
-      { id: "user 1", tz: "America/Denver" },
-      { id: "user 2", tz: "America/New_York" },
-      { id: "user 3", tz: "America/Chicago" },
-      { id: "user 4", tz: "America/Anchorage" },
+      // puerto rico does not observe daylight savings time and is therefore
+      // excluded from the date-unaware tests to avoid failing half of the year
+      { id: "user ny", tz: "America/New_York" },
```
Comment (mgwalker) [inline src/scripts/timezone.test.js:65]: THIS IS SO MUCH NICER! 😍

File: src/scripts/timezone.test.js [hunk 2]
```diff
@@ -127,67 +131,101 @@ describe("Handy Tau-bot timezone conversions", () => {
...
 
       expect(slack.postEphemeralMessage).toHaveBeenCalledWith(
-        responseFor("user 1", "2:00 pm")
+        responseFor("user dn", "2:00 pm")
       );
 
       // Not testing filtering here, so trust the previous tests.
     });
 
+    it("correctly includes AM/PM when crossing noon", async () => {
```
Comment (mgwalker) [inline src/scripts/timezone.test.js:183]: This is a good addition. These tests should make it safer to eventually migrate away from moment.js.

File: src/scripts/timezone.test.js [hunk 3]
```diff
@@ -196,5 +234,27 @@ describe("Handy Tau-bot timezone conversions", () => {
...
+      expect(slack.postEphemeralMessage).not.toHaveBeenCalled();
+    });
+
+    // currently failing but may be worth investigating in future
+    it.skip("correctly converts 24 hour time", async () => {
+      message.event.text = "23:42 cst";
+      await handler(message);
+
+      expect(slack.postEphemeralMessage).toHaveBeenCalledWith(
+        responseFor("user ny", "12:42 est")
```
Comment (mgwalker) [inline src/scripts/timezone.test.js:251]:
```suggestion
        responseFor("user ny", "12:42")
```

The response wouldn't include a timezone. I think that might be why this test is failing.

File: src/scripts/timezone.test.js [hunk 4]
```diff
@@ -196,5 +234,27 @@ describe("Handy Tau-bot timezone conversions", () => {
...
+      );
+
+      expect(slack.postEphemeralMessage).toHaveBeenCalledWith(
+        responseFor("user la", "9:42 pst")
```
Comment (mgwalker) [inline src/scripts/timezone.test.js:255]:
```suggestion
        responseFor("user la", "9:42")
```

**Review by mgwalker — APPROVED** [review]
Nice fix, and the updates to the test are great! Merge it whenever you're ready.

File: src/scripts/timezone.test.js (hunk 3 above)
Comment (jduss4) [reply src/scripts/timezone.test.js]: Yep, that was it!  Thanks for the catch :)

**Pull request merged**